codec-http/src/main/java/io/netty/handler/codec/http/HttpContentCompressor.java

```diff
@@ -239,6 +251,11 @@ protected Result beginEncode(HttpResponse httpResponse, String acceptEncoding) t
                         new EmbeddedChannel(ctx.channel().id(), ctx.channel().metadata().hasDisconnect(),
                                 ctx.channel().config(), new BrotliEncoder(brotliOptions.parameters())));
             }
+            if (targetContentEncoding.equals("zstd")) {
```

Comment:
Time to use a `Map` lookup instead of four `if`s?

Reply:
How about `switch`?

Reply:
If it compiles? Whichever way that makes things cleaner :slightly_smiling_face:

Reply:
However, the string class of switch statement has only been supported since java7. Is this ok?

Reply:
I guess not because Netty 4.1 still at an older version of Java IIRC. Perhaps my comment is beyond the scope of this PR. What do you think about just adding a comment about the improvements we could make in this perspective?

Reply:
How about adding an enum, we can judge contentEncoding in the enum according to the `valueOf` method, like this:

```
public interface RoleService {
     void approve();
}
```

```
public enum RoleServiceEnum implements RoleService {
     ROLE_LEADER(){
         @Override
         public void approve() {
             System.out.println("leader doing sth");
         }
     },
 
     ROLE_SERVICE_OWNER(){
         @Override
         public void approve() {
             System.out.println("so doing sth");
         }
     },
 
     ROLE_PROJECT_MANAGER(){
         @Override
         public void approve() {
             System.out.println("pm doing sth");
         }
     },
 
     ROLE_SUPERVISOR(){
         @Override
         public void approve() {
             System.out.println("supervisor doing sth");
         }
     };
}
```

```
public void judge(String role){
         RoleServiceEnum.valueOf(role).approve();
     }
```
Or can add a `match` method to the enum, like this:

```
public static ChannelRuleEnum match(String name) {
        ChannelRuleEnum[] values ​​= ChannelRuleEnum.values();
        for (ChannelRuleEnum value: values) {
            if (value.name.equals(name)) {
                return value;
            }
        }
        return null;
    }
```
How does it sound?

Reply:
Yeah, I thought something like that as well, but it seemed like too much work for you. If you are fine with that route, I'd be more than happy for sure. :bow:

Reply:
I can do it, don't worry about this

Reply:
Awesome, @skyguard1. Much appreciated!

Reply:
I would do this as a separate PR ... to keep the scope as small as possible

Reply:
Looking forward to seeing this pr!

Reply:
actually I was hoping you would provide a PR :) @skyguard1

Reply:
Maybe i can do this next week